Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Lights/DirectionalLight.azsli

```diff
@@ -50,21 +67,30 @@ void ApplyDirectionalLights(Surface surface, inout LightingData lightingData)
         // [GFX TODO][ATOM-2012] care of multiple directional light
         // Currently shadow check is done only for index == shadowIndex.
         float currentLitRatio = 1.0f;
-        float currentBackShadowRatio = 1.0f;
+
+        // Transmission distance from current light inside object (default non-influential values)
+        // o_transmission_mode == NONE is not taken into account because GetBackLighting already ignores this case 
+        float currentTransmissionDistance = (o_transmission_mode == TransmissionMode::ThickObject) ? 0.0f : 9999.0f;
         if (o_enableShadows)
         {
-            currentLitRatio = (index == shadowIndex) ? litRatio : 1.;
-            
-            currentBackShadowRatio = 1.0 - currentLitRatio;
-            if (o_transmission_mode == TransmissionMode::ThickObject)
+            bool activeLight = index == shadowIndex;
+            currentLitRatio = activeLight ? litRatio : 1.;
+            if (activeLight)
             {
-                currentBackShadowRatio = (index == shadowIndex) ?  backShadowRatio : 0.;
+                // Transmission distance (add contribution only if current directional light is the active one for shadows)
+                currentTransmissionDistance = transmissionDistance;
             }
         }
         
         lightingData.diffuseLighting += GetDiffuseLighting(surface, lightingData, light.m_rgbIntensityLux, dirToLight) * currentLitRatio;
         lightingData.specularLighting += GetSpecularLighting(surface, lightingData, light.m_rgbIntensityLux, dirToLight) * currentLitRatio;
-        lightingData.translucentBackLighting += GetBackLighting(surface, lightingData, light.m_rgbIntensityLux, dirToLight, currentBackShadowRatio);
+        
+        float3 backLighting = GetBackLighting(surface, lightingData, light.m_rgbIntensityLux, dirToLight, currentTransmissionDistance);
+        
+        // Attenuation applied to hide artifacts due to low-res shadow maps (might need some work in the future)
+        float attenuation = 1.0 / pow(max(1.0, camToSurfDist/2.0), lightingData.distanceAttenuation + 1.0);
```

Comment:
I suggest moving this and the attenuation parameter into `GetBackLighting` to reduce some code duplication and also make it easier to adjust the backlighting model in one place in the future

Reply:
Makes sense! I've added a new parameter to the `GetBackLighting` function, `attenuationDistance` that contains the computed distance to use for attenuating (which might be different for each light type, e.g. distance to light in point or disk lights, and distance to the camera in directional lights).  
This way, the attenuation parameter computation can be moved into `GetBackLighting`. 
(done in https://github.com/o3de/o3de/pull/6428/commits/eda405f5f1d3ca785f61ea9ae1f4b9fb3ea54c0a)